### PR TITLE
Docs: Improve Docker documentation

### LINKED ISSUE #1910: Docs: Improve Docker documentation
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

The [Quickstart w. Docker](https://docs.medusajs.com/quickstart/quick-start-docker/) documentation needs improvement because:

1. It doesn't follow our style guide
2. The information might not be up to date or might not work
3. It's not in the sidebar

### How can this issue be resolved?

The documentation should:

1. Follow our style guide as indicated in our [contribution guidelines](https://docs.medusajs.com/contribution-guidelines)
2. Make sure the information in it are up to date and work
3. If there are any necessary steps needed to integrate with the storefront or admin that are different from the usual steps of integrating with Medusa, they should be specified.
4. A link to the documentation should be added in the sidebar.

### Requirement

Whoever wants to work on this issue must be experienced with Docker

### Are you interested in working on this issue?

- [ ] I would like to fix this issue

### PR DESCRIPTION
**Pull request by Omar8345**
I have made the doc to follow **the guidelines**, **updated commands (Docker)**, **added the doc to sidebar**.

Closes #1910 

@shahednasser Please check

### CODE REVIEW

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your work @Omar8345 . I have left some comments below, but I have some additional comments:

- Add a Prerequisites section that Docker is required with a link to install it. (https://docs.docker.com/get-docker/)
- Remove Step 4 (configurations part)
- Add a Troubleshooting Section after the Testing section saying if the error `Error response from daemon: Ports are not available: exposing port TCP 0.0.0.0:5432 -> 0.0.0.0:0: listen tcp 0.0.0.0:5432: bind: address already in use` occurs they can change the port of postgres in `docker-compose.yml` to the following:

```yml
  postgres:
    ports:
      - "5433:5432"
```

- Copy the Additional Steps and What's Next sections from the quickstart guide into here
- There are a lot of changes in the `sidebar.js` file. This is likely due to configurations in your editor. Can you copy the current file in the `master` branch and just make one change to add the link?

File: docs/content/quickstart/quick-start-docker.md [hunk 1]
```diff
@@ -50,14 +50,14 @@ SENDGRID_API_KEY="some_sendgrid_key"
...
 
-We will use docker-compose and Docker to start up our development environment. Running the above command will do the following:
+You will use `docker compose` and Docker to start up your development environment. Running the above command will do the following:
 
-1. Build images for our Medusa project, a PostgreSQL database and a Redis server
-2. Run migrations for our newly created database
-3. Seed our database with some entities, that will allow us to easily get started.
+1. Build images for your Medusa project, a PostgreSQL database and a Redis server
+2. Run migrations for your newly created database
+3. Seed your database with some entities, that will allow us to easily get started.
```
[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/quickstart/quick-start-docker.md:60]: Actually remove number 3

File: docs/content/quickstart/quick-start-docker.md [hunk 2]
```diff
@@ -67,11 +67,11 @@ We will use docker-compose and Docker to start up our development environment. R
    - A product called Cool Test Product
    - A variant of that product that costs 195 EUR
 
-Once done, our server will be accessible at `http://localhost:9000`.
+Once done, your server will be accessible at `http://localhost:9000`.
```
[the remark this answers is not in the document]
Comment (Omar8345) [reply docs/content/quickstart/quick-start-docker.md:70]: LGFM!

File: docs/content/quickstart/quick-start-docker.md [hunk 3]
```diff
@@ -1,34 +1,34 @@
-# Quickstart w. Docker
+# Quickstart with Docker
 
-This quick start is intended for developers, that have already configured their local development environment and familiarised them selves with all the technologies and frameworks used throughout the Medusa eco-system.
+This quick start is intended for developers, assuming that you have already configured your local development environment and familiarised yourself with all the technologies and frameworks used throughout the Medusa eco-system.
```
[the remark this answers is not in the document]
Comment (shahednasser) [reply docs/content/quickstart/quick-start-docker.md:3]: @Omar8345 why is this resolved?

**Comment by shahednasser** [comment]
@Omar8345 I can see that you resolved comments without making the changes in those comments. Please make those changes then resolve the comments.

**Pull request closed without merging**